### PR TITLE
Multi word expressions feature

### PR DESCRIPTION
**Pull request by bbonf**
MWE search mostly works now.
Further changes will be made in another branch/PR

### CODE REVIEW

File: web-ui/src/app/components/step/query-set/query-set.component.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
+<div class="query" *ngFor="let query of queryset; let idx = index">
+  <h4>Query {{ idx + 1 }}: {{ query.description }}</h4>
+  <div>
+    <grt-xpath-editor
+        [value]="query.xpath">
+    </grt-xpath-editor>
+  </div>
+  <button class="button" (click)="sendQuery(idx)">Search</button>
+</div>
```
Comment (tijmenbaarda) [inline web-ui/src/app/components/step/query-set/query-set.component.html:10]: Is this component used anywhere?

File: backend/search/views.py [hunk 2]
```diff
@@ -45,11 +71,23 @@ def search_view(request):
...
+    behaviour = data.get('behaviour', {})
+
     if is_analysis:
         maximum_results = settings.MAXIMUM_RESULTS_ANALYSIS
     else:
         maximum_results = settings.MAXIMUM_RESULTS
 
+    use_superset = behaviour.get('supersetXpath') is not None
+    # TODO: find a nicer way for running just the superset query
+    use_superset = use_superset and behaviour['supersetXpath'] != xpath
```
Comment (tijmenbaarda) [inline backend/search/views.py:83]: Use one assignment for this?

File: backend/search/views.py [hunk 3]
```diff
@@ -45,11 +71,23 @@ def search_view(request):
...
     variables = data.get('variables', [])
+    behaviour = data.get('behaviour', {})
+
     if is_analysis:
         maximum_results = settings.MAXIMUM_RESULTS_ANALYSIS
     else:
         maximum_results = settings.MAXIMUM_RESULTS
 
+    use_superset = behaviour.get('supersetXpath') is not None
+    # TODO: find a nicer way for running just the superset query
```
Comment (tijmenbaarda) [inline backend/search/views.py:82]: I am not really sure what you mean and why you would want that

File: backend/search/views.py [hunk 4]
```diff
@@ -45,11 +71,23 @@ def search_view(request):
...
+    use_superset = use_superset and behaviour['supersetXpath'] != xpath
+
+    should_expand_index = behaviour.get('expandIndex', False)
+    if use_superset:
+        subset_xpath = xpath
+        xpath = behaviour['supersetXpath']
+
+
```
Comment (tijmenbaarda) [inline backend/search/views.py:90]: Double blank line

File: backend/search/views.py [hunk 5]
```diff
@@ -17,11 +17,37 @@
...
 
 def run_search(query_obj) -> None:
     query_obj.perform_search()
 
 
+def filter_subset_results(results, xpath, should_expand_index):
+    out = []
+    for result in results:
+        sent = ET.fromstring(result['xml_sentences'])
```
Comment (tijmenbaarda) [inline backend/search/views.py:36]: I was a bit confused by the variable name `sent`. Change to `sentences`?

File: backend/search/views.py [hunk 6]
```diff
@@ -45,11 +71,23 @@ def search_view(request):
...
     if is_analysis:
         maximum_results = settings.MAXIMUM_RESULTS_ANALYSIS
     else:
         maximum_results = settings.MAXIMUM_RESULTS
 
+    use_superset = behaviour.get('supersetXpath') is not None
+    # TODO: find a nicer way for running just the superset query
+    use_superset = use_superset and behaviour['supersetXpath'] != xpath
+
+    should_expand_index = behaviour.get('expandIndex', False)
```
Comment (tijmenbaarda) [inline backend/search/views.py:85]: It seems this is only used if `use_superset` is True, so maybe move inside if block. Also, a comment on what `use_superset` and `should_expand_superset` do would be useful

**Review by tijmenbaarda — COMMENTED** [review]
Great! I did not understand everything about MWE 100% but I did my best. I made some further changes to make context toggling work again, because the way that works had changed in the meantime.

File: backend/mwe/models.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+class XPathQuery(models.Model):
+    class Meta:
+        verbose_name = 'XPath Query'
+        verbose_name_plural = 'XPath Queries'
+
+    canonical = models.ForeignKey(CanonicalForm, on_delete=models.PROTECT)
+    xpath = models.TextField()
+    description = models.CharField(max_length=200)
+    rank = models.PositiveSmallIntegerField()
```
[the remark this answers is not in the document]
Comment (bbonf) [reply backend/mwe/models.py:27]: I agree with the comment, but I'm not sure these fields will stay. I'd rather do another documentation-focused PR at a later stage

**Pull request merged**